### PR TITLE
Support more model outputs for BERT/ERNIE/RoBERTa

### PR DESCRIPTION
**Pull request by guoshengCS**
### PR types

New features 
### PR changes

APIs
### Description

Support more model outputs for BERT. The default behavior is the same as before for compatibility.

Import more output structures refer to HF.

```python
    BaseModelOutputWithPastAndCrossAttentions,
    BaseModelOutputWithPoolingAndCrossAttentions,
    SequenceClassifierOutput,
    TokenClassifierOutput,
    QuestionAnsweringModelOutput,
    MultipleChoiceModelOutput,
    MaskedLMOutput,
    ModelOutput,
```

Usage:

```python
import paddle
import paddlenlp
from paddlenlp.transformers import AutoModel, AutoTokenizer, AutoModelForTokenClassification, AutoModelForPretraining

paddle.seed(123)
model_name = "bert-base-uncased"
model_name = "ernie-1.0-base-zh"
model_name = "hfl/roberta-wwm-ext"
#model = AutoModel.from_pretrained(model_name)
model = AutoModelForTokenClassification.from_pretrained(model_name)
#model = AutoModelForPretraining.from_pretrained(model_name)
model.eval()
tokenizer = AutoTokenizer.from_pretrained(model_name)

inputs = tokenizer("欢迎使用百度飞桨!")
inputs = {k:paddle.to_tensor([v]) for (k, v) in inputs.items()}
print(inputs)
outputs = model(**inputs, return_dict=True, output_hidden_states=True, output_attentions=False)
print(outputs)
```

### CODE REVIEW

File: paddlenlp/transformers/model_outputs.py [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+
+
+import inspect
+import dataclasses
+from collections import OrderedDict, UserDict
+from collections.abc import MutableMapping
+from contextlib import ExitStack
+from dataclasses import fields, dataclass
+from enum import Enum
+from typing import Any, ContextManager, List, Tuple, Optional
```
Comment (yingyibiao) [inline paddlenlp/transformers/model_outputs.py]: 这里的 import 是否应该挪到文件开头位置

File: paddlenlp/transformers/model_outputs.py [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+                            output_attentions=output_attentions)
+        if isinstance(layer_outputs, tuple):
+            output = layer_outputs[0]
+            outputs = layer_outputs[1:]
+        else:
+            output = layer_outputs
+            outputs = None
+
+        if output_hidden_states:
+            all_hidden_states.append(output)
```
Comment (yingyibiao) [inline paddlenlp/transformers/model_outputs.py]: HF 中的 all_hidden_states 包含了 embedding_output，这里的实现相比会少这个 Tensor。

File: paddlenlp/transformers/model_outputs.py (hunk 1 above)
Comment (guoshengCS) [reply paddlenlp/transformers/model_outputs.py]: Done

File: paddlenlp/transformers/model_outputs.py (hunk 2 above)
Comment (guoshengCS) [reply paddlenlp/transformers/model_outputs.py]: Done. Thanks

**Review by yingyibiao — APPROVED** [review]
LGTM

**Pull request merged**